modules/compute/src/main/java/org/apache/ignite/internal/compute/task/DelegatingTaskExecution.java

```diff
@@ -60,4 +62,9 @@ public CompletableFuture<R> resultAsync() {
     public CompletableFuture<@Nullable Boolean> changePriorityAsync(int newPriority) {
         return delegate.thenCompose(execution -> execution.changePriorityAsync(newPriority));
     }
+
+    @Override
+    public @Nullable Marshaller<R, byte[]> resultMarshaller() {
+        return delegate.join().resultMarshaller();
```

Comment:
Delegate future is supposed to be completed at this point, right? Let's add an assertion.